R/db-interface.R

```diff
@@ -242,19 +263,31 @@ check_naming <- function(table_names, dm_table_names) {
   }
 }
 
-db_append_table <- function(con, remote_table, table) {
-  if (nrow(table) == 0) {
+db_append_table <- function(con, remote_table, table, progress, top_level_fun = "copy_dm_to") {
+  if (nrow(table) == 0 || ncol(table) == 0) {
     return(invisible())
   }
 
   if (is_mssql(con)) {
-    # https://github.com/r-dbi/odbc/issues/480
-    values <- as_tibble(map(table, map_chr, dbplyr::escape, con = con))
-    # chunks of 1000
-    idx <- as.integer((seq_len(nrow(values)) + 999L) / 1000L)
-    split <- vec_split(values, idx)
-    sql <- map_chr(split$val, ~ DBI::sqlAppendTable(con, DBI::SQL(remote_table), as.list(.x), row.names = FALSE))
-    walk(sql, ~ DBI::dbExecute(con, .x, immediate = TRUE))
+    # FIXME: Make adaptive
+    chunk_size <- 1000L
+    n_chunks <- ceiling(nrow(table) / chunk_size)
+
+    ticker <- new_ticker(
+      paste0("inserting into ", remote_table),
+      n = n_chunks,
+      progress = progress,
+      top_level_fun = top_level_fun
+    )
+
+    walk(seq_len(n_chunks), ticker(~ {
+      end <- .x * chunk_size
+      idx <- seq2(end - (chunk_size - 1), min(end, nrow(table)))
+      values <- as_tibble(map(table[idx, ], dbplyr::escape, parens = FALSE, collapse = NULL, con = con))
```

Comment:
I believe the `as_tibble()`-call is redundant, since you are converting `values` to a list again the only time you're using it later on

Reply:
Good catch!